### ISSUE TITLE
Disposable factory - expected to see a warning, but there wasn't 

### ISSUE THREAD

**Issue by galarlo** [DotNetAnalyzers/IDisposableAnalyzers#521]
I have a factory class which creates `IDisposable`s. I expect to see a warning when using the factory without disposing it's result.  
However, ther is no warning.

Is there anything I can do to make a warning appear?

Here's a simplified code snippet:

```csharp
public interface IDisposableFactory
{
    IDisposable Create();
}

public class Example
{
    void Example(IDisposableFactory factory)
    {
         IDisposable disposable = factory.Create(); // why no warning?
    }
}
```
